src/drive/web/modules/drive/FileOpenerExternal.jsx

```diff
@@ -70,11 +78,13 @@ const getFileId = ownProps => {
 
 FileOpener.PropTypes = {
   router: PropTypes.shape({
+    push: PropTypes.func.isRequired,
     params: PropTypes.shape({
       fileId: PropTypes.string.isRequired
     }).isRequired
   }).isRequired,
-  fileId: PropTypes.number
+  fileId: PropTypes.number,
+  withCloseButtton: PropTypes.bool
 }
 
-export default translate()(FileOpener)
+export default translate()(withRouter(FileOpener))
```

Comment:
This file is only used when we open a document from a deeplink, right?

Reply:
No, it's also used in the OPEN file intent view.

Reply:
Oh hang on that's a good point, there might not be a router in that case. Not that it would be used, but I still need to check this.

Reply:
All good, I can confirm it works in both cases!